components/media/MediaResolver.vue

```diff
@@ -0,0 +1,40 @@
+<template>
+  <component
+    :is="resolveComponent"
+    v-if="src"
+    :src="src"
+    :poster="poster || '/placeholder.webp'"
+    :description="description"
+    :preview="preview"
+    :availableAnimations="availableAnimations" />
+</template>
+
+<script lang="ts" setup>
+import { computed, defineAsyncComponent } from '#app'
+import { resolveMedia } from '@/utils/gallery/media'
+
+const props = defineProps<{
+  src?: string
+  mimeType?: string
+  poster?: string
+  description?: string
+  availableAnimations?: string[]
+  preview?: boolean
+}>()
+
+const SUFFIX = 'Media'
+const components = {
+  VideoMedia: defineAsyncComponent(() => import('./type/VideoMedia.vue')),
+  ImageMedia: defineAsyncComponent(() => import('./type/ImageMedia.vue')),
+  AudioMedia: defineAsyncComponent(() => import('./type/AudioMedia.vue')),
+  JsonMedia: defineAsyncComponent(() => import('./type/JsonMedia.vue')),
+  ModelMedia: defineAsyncComponent(() => import('./type/ModelMedia.vue')),
+  IFrameMedia: defineAsyncComponent(() => import('./type/IFrameMedia.vue')),
+  ObjectMedia: defineAsyncComponent(() => import('./type/ObjectMedia.vue')),
+  Media: defineAsyncComponent(() => import('./type/UnknownMedia.vue')),
+}
+
+const resolveComponent = computed(() => {
```

Comment:
```suggestion
const resolveComponent = $computed(() => {
```

might work without import?